### DISCUSSION TITLE
Hebrew/RTL support

### DISCUSSION THREAD

**Discussion by elikalins** [codefori/vscode-ibmi#1876] [Q&A]
Hello everyone,
I am experiencing difficulties with Hebrew support in my editor. The words get jumbled up when there are both English and Hebrew letters. 
5250 (original) :
![image](https://github.com/codefori/vscode-ibmi/assets/57211167/ed36d8cb-5861-43ad-8ff1-b69b06314c06)
Vs code :
![image](https://github.com/codefori/vscode-ibmi/assets/57211167/d46df2c5-effe-4508-b783-6dc1ef6a3b9e)
Do you have any suggestions for how to solve this problem?

Thank you.

**Comment by worksofliam** [comment]
Hey @elikalins!

So we can test with this: what is the CCSID of the source file?

Thanks,
Liam

**Reply by worksofliam** [reply]
Can you also share if you have source dates enabled or disabled?

**Reply by elikalins** [reply]
hi @worksofliam  , 
CCSID of  the files is 424 
source dates - disabled

**Reply by worksofliam** [reply]
@elikalins 

Is it possible you can share the hex of that line that contains the RTL characters? Thanks!

**Reply by elikalins** [reply]
x'E2D5C4D1D4C1C9D3404040E2E4C2D1C5C3E34D7D4040405551586445404468695640714668516940714651586440484644407D5D404E' 
![image](https://github.com/codefori/vscode-ibmi/assets/57211167/e2d9e4cb-19c1-4568-a764-178708cec594)

**Reply by worksofliam** [reply]
@elikalins To confirm, that hex is the same as the source member you shared, right? 

![image](https://github.com/codefori/vscode-ibmi/assets/3708366/89d89b2f-9a4d-40a3-ab50-6622a6342ac0)

**Reply by worksofliam** [reply]
It doesn't look like `FD` (`\u200E`) is in your hex string!

**Reply by worksofliam** [reply]
Hey @elikalins !!

I recreated it!!

```
cl: crtsrcpf liama/hebrew ccsid(424);
cl: ADDPFM FILE(LIAMA/HEBREW) MBR(HEBREW) TEXT('Hebrew');
insert into liama/hebrew values (
  0, 0, x'E2D5C4D1D4C1C9D3404040E2E4C2D1C5C3E34D7D4040405551586445404468695640714668516940714651586440484644407D5D404E'
);

select * from liama/hebrew;
```

<img width="986" alt="image" src="https://github.com/codefori/vscode-ibmi/assets/3708366/8e2d4eed-8bad-44e4-9652-9d5b829fe211">

**Reply by worksofliam** [reply]
Here is how I recreated it:

```
cl: crtsrcpf liama/hebrew ccsid(424);
cl: ADDPFM FILE(LIAMA/HEBREW) MBR(HEBREW) TEXT('Hebrew');
insert into liama/hebrew values (
  0, 0, x'E2D5C4D1D4C1C9D3404040E2E4C2D1C5C3E34D7D4040405551586445404468695640714668516940714651586440484644407D5D404E'
);

cl: CPYTOSTMF FROMMBR('/QSYS.LIB/LIAMA.LIB/HEBREW.FILE/HEBREW.MBR') TOSTMF('/tmp/vscodetemp-O_0p9XHiyZ') STMFOPT(*REPLACE) STMFCCSID(1208) DBFCCSID(62245);

select * from liama/hebrew;
```

**Comment by worksofliam** [comment]
Related issue #1781
